src/passes/DeadArgumentElimination.cpp

```diff
@@ -242,17 +254,18 @@ struct DAE : public Pass {
     DAEFunctionInfoMap infoMap;
     // Ensure they all exist so the parallel threads don't modify the data
     // structure.
-    ModuleUtils::iterDefinedFunctions(
-      *module, [&](Function* func) { infoMap[func->name]; });
+    for (auto& func : module->functions) {
+      infoMap[func->name];
+    }
     // Check the influence of the table and exports.
     for (auto& curr : module->exports) {
       if (curr->kind == ExternalKind::Function) {
-        infoMap[curr->value].hasUnseenCalls = true;
+        infoMap[curr->value].hasUnseenCalls.store(true);
```

Comment:
Hmm, I tried this, and the docs say it should work... but I get race conditions that way.

You know more about atomics than me. Is there some gotcha here? Maybe just with loads or just with stores?

Reply:
Or maybe just with certain versions of C++? (I found no mention of gotchas in any docs, though)

Reply:
No, that's very strange. What race condition symptoms did you see?

Reply:
It looks like changing from `load/store` to plain `=` does not change the object code, so that is fine as the docs say. I think what I also did was assume that the `std::atomic<bool>` is initialized to false, but when I remove that line, the object code does change. So I guess I mixed the two things together. And it's fine to initialize it with either `.store` or `=`.